src/main.rs

```diff
@@ -179,6 +179,7 @@ fn generate_peer_threads(
                         request_blocks(Arc::clone(&t), &mut c);
                     }
                 }
+                Err(e) => println!("connection err: {:?}", e),
```

Comment:
I expect the connection to close here because the connection will be dropped and so will the underlying tcp stream.